tapchannel/aux_invoice_manager.go

```diff
@@ -227,6 +252,59 @@ func (s *AuxInvoiceManager) priceFromQuote(rfqID rfqmsg.ID) (
 	}
 }
 
+// rfqPeerFromScid attempts to match the provided scid with a negotiated quote,
+// then it returns the RFQ peer's node id.
+func (s *AuxInvoiceManager) rfqPeerFromScid(scid uint64) (route.Vertex, error) {
+	acceptedBuyQuotes := s.cfg.RfqManager.PeerAcceptedBuyQuotes()
+	acceptedSellQuotes := s.cfg.RfqManager.LocalAcceptedSellQuotes()
+
+	buyQuote, isBuy := acceptedBuyQuotes[rfqmsg.SerialisedScid(scid)]
+	sellQuote, isSell := acceptedSellQuotes[rfqmsg.SerialisedScid(scid)]
+
+	switch {
+	case isBuy:
+		return buyQuote.Peer, nil
+
+	case isSell:
+		return sellQuote.Peer, nil
+
+	default:
+		return route.Vertex{},
+			fmt.Errorf("no peer found for RFQ SCID %d", scid)
+	}
+}
+
+// isAssetInvoice checks whether the provided invoice is an asset invoice. This
+// method checks whether the routing hints of the invoice match those created
+// when generating an asset invoice, and if that's the case we then check that
+// the scid matches an existing quote.
+func (s *AuxInvoiceManager) isAssetInvoice(invoice *lnrpc.Invoice) bool {
+	hints := invoice.RouteHints
+
+	if len(hints) != 1 {
+		return false
+	}
+
+	hint := hints[0]
+	if len(hint.HopHints) != 1 {
```

Comment:
Should this be a minimum of 1, instead of exactly 1? Seems like this would cause issues if we wanted to use hop hints for other reasons with this payment.

Also, AFAICT we only need to inspect one hint (the 'last' hint) for this use case.

Reply:
Right, we could have multiple inbound channels with the same asset. So I guess we should just check if _one_ of the hop hints has an SCID that corresponds to an RFQ ID.

Reply:
Updated to embrace various routing hints / hop hints

we now look for at least 1 hop hint that related to a valid rfq